src/Nethermind/Nethermind.Evm/IntrinsicGasCalculator.cs

```diff
@@ -57,6 +57,11 @@ private static long DataCost(Transaction transaction, IReleaseSpec releaseSpec)
                 }
             }
 
+            if (transaction.IsContractCreation && releaseSpec.IsEip3860Enabled)
+            {
+                dataCost += EvmPooledMemory.Div32Ceiling((ulong)transaction.Data.Length) * GasCostOf.InitCodeWord;
```

Comment:
why ulong cast and not directly to UInt256?

Reply:
In all the occurances I found when Div32Ceiling was called for an int parameter, it was cast to ulong.

Reply:
Please cast directly to UInt256